src/libs/ReportActionsUtils.js

```diff
@@ -37,6 +39,19 @@ Onyx.connect({
     },
 });
 
+const allTransactions = {};
+Onyx.connect({
+    key: ONYXKEYS.COLLECTION.TRANSACTION,
+    callback: (actions, key) => {
+        if (!key || !actions) {
+            return;
+        }
+
+        const transactionID = CollectionUtils.extractCollectionItemID(key);
+        allTransactions[transactionID] = actions;
+    },
+});
+
```

Comment:
I think we should avoid adding this here, we are pulling all the transactions in the `TransactionUtils` here https://github.com/Expensify/App/blob/61ab4bcf100500ce49bf1e2dfaa128f9511ea941/src/libs/TransactionUtils.js#L10-L21 and `getReportPreviewTransactionsWithReceipts` should be moved there as it does not new allReportActions. This will keep it cleaner imho